src/logic/issues.py

```diff
@@ -0,0 +1,110 @@
+import enum
+import re
+import typing
+from dataclasses import dataclass, field
+
+import redis
+from github.Issue import Issue as GhIssue
+from github.Repository import Repository as GhRepository
+
+from ..github_auth import get_repo_client
+from ..repo_config import RepoConfig
+from ..settings import Settings, log
+from . import models
+
+
+class IssueAction(str, enum.Enum):
+    # TODO: Use StrEnum in Python 3.11
+    OPENED = 'opened'
+    REOPENED = 'reopened'
+
+
+def process_issue(*, event: models.IssueEvent, settings: Settings) -> tuple[bool, str]:
+    """Processes an issue in the repo
+
+    Performs following actions:
+    - assigns new issues and reopen to a next person in the assignees list
+
+    TODO:
+    - assign reopened issue to the assignee selected before
+    - use "can confirm" magic comment from a contributor to change labels (remove an `unconfirmed` label)
+    - use "please update" magic comment to reassign to the author
+    - reassign from the author back to contributor after any author's comment
+    """
+    if event.action not in iter(IssueAction):
+        return False, f'Unapplicable issue action "{event.action}"'
+
+    with get_repo_client(event.repository.full_name, settings) as gh_repo:
+        gh_issue = gh_repo.get_issue(event.issue.number)
+        config = RepoConfig.load(issue=gh_issue, settings=settings)
+
+        log(f'{event.issue.user} ({event.action}): #{event.issue.number}')
+
+        label_assign = LabelAssign(
+            gh_issue=gh_issue,
+            gh_repo=gh_repo,
+            action=IssueAction,
+            author=event.issue.user,
+            repo_fullname=event.repository.full_name,
+            config=config,
+            settings=settings,
+        )
+
+        label_assign.assign_new()
+
+    return True, ''
+
+
+@dataclass(kw_only=True)
+class LabelAssign:
+    # for example "Selected Assignee: @samuelcolvin"
+    ASSIGNEE_REGEX: typing.ClassVar[re.Pattern] = re.compile(r'selected[ -]assignee:\s*@([\w\-]+)$', flags=re.I)
+
+    gh_issue: GhIssue
+    gh_repo: GhRepository
+    action: IssueAction
+    author: str
+    repo_fullname: str
+    config: RepoConfig
+    settings: Settings
+    assignees: list[str] = field(init=False)
+
+    def __post_init__(self):
+        self.assignees = self.config.assignees or [user.login for user in self.gh_repo.get_collaborators()]
+
+    def assign_new(self):
+        assignee = self._select_assignee()
+        self._assign_user(assignee)
+        self._add_label(self.config.unconfirmed_label)
```

Comment:
what is the example use here of changing labels?

Reply:
A new issue gets `unconfirmed` label. After confirming contributor removes `unconfirmed` label.

There is a TODO in `process_issue` doc string suggesting adding a reaction on a magic comment like "can confirm" that should change `unconfirmed` label to `confirmed`.